test/rectangle_map_test.dart

```diff
@@ -1,44 +1,48 @@
 import 'package:test/test.dart';
 import 'package:tiled/tiled.dart';
-import 'package:xml/xml.dart';
 import 'dart:io';
 
 void main() {
-  XmlElement xmlRoot;
+  TiledMap map;
 
   setUp(() {
     return File('./test/fixtures/rectangle.tmx').readAsString().then((xml) {
-      xmlRoot = XmlDocument.parse(xml).rootElement;
+      map = TileMapParser.parseTmx(xml);
     });
   });
 
   group('rectangle', () {
     test('rectangle map works from layer', () {
-      final layerNode = xmlRoot.findAllElements('layer').first;
-      final layer = Layer.fromXML(layerNode);
+      final layer = map.getLayerByName('Tile Layer 1');
 
-      expect(layer.tileMatrix[0], equals([1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]));
-      expect(layer.tileMatrix[1], equals([0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]));
-      final flip = layer.tileFlips[1].last;
-      expect(flip.vertically, equals(true));
-      expect(flip.diagonally, equals(true));
-      expect(layer.tileMatrix[2], equals([0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0]));
-      expect(layer.tileMatrix[3], equals([0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0]));
-      expect(layer.tileMatrix[4], equals([0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0]));
-      expect(layer.tileMatrix[5], equals([0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0]));
-      expect(layer.tileMatrix[6], equals([0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0]));
-      expect(layer.tileMatrix[7], equals([0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0]));
-      expect(layer.tileMatrix[8], equals([0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0]));
-      expect(layer.tileMatrix[9], equals([0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0]));
+      expect(
+          layer.tileIDMatrix[0], equals([1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]));
+      expect(
+          layer.tileIDMatrix[1], equals([0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]));
+      expect(layer.tileFlips[1].last.vertically, equals(true));
+      expect(layer.tileFlips[1].last.diagonally, equals(true));
+      expect(
+          layer.tileIDMatrix[2], equals([0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0]));
+      expect(
+          layer.tileIDMatrix[3], equals([0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0]));
+      expect(
+          layer.tileIDMatrix[4], equals([0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0]));
+      expect(
+          layer.tileIDMatrix[5], equals([0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0]));
+      expect(
+          layer.tileIDMatrix[6], equals([0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0]));
+      expect(
+          layer.tileIDMatrix[7], equals([0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0]));
+      expect(
+          layer.tileIDMatrix[8], equals([0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0]));
+      expect(
+          layer.tileIDMatrix[9], equals([0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0]));
```

Comment:
Minor comment, these should have a trailing comma since it is split on two lines.

Reply:
we can do that as a followup